### ISSUE TITLE
Dimension 0 in both shapes must be equal, but are 4 and 32 for 'Assign_1' (op: 'Assign') with input shapes: [4,4,3,64], [32,3,2,2].

### ISSUE THREAD

**Issue by Ham72** [keras-team/keras#8682]
I am using convolutional network with filter size 64. I want to train the model for 100 Epoch. It works fine for 10-12 Epoch, after that gives the the error mentioned in title. Even if I run it for epoch=1, after completing the epoch it gives the same error.

The dimension of data is 
Train data(40000, 32, 32, 3)
label train data(40000, 10)
Val data(10000, 32, 32, 3)
label Val data(10000, 10)

Code is given below.
input_shape = (32,32,3)
model = Sequential() 
model.add(Conv2D(64, 4, strides=1, activation='relu', input_shape = input_shape)) 
model.add(MaxPooling2D(pool_size=2))
model.add(Flatten())
model.add(Dense(50, activation='relu'))
model.add(Dense(10, activation= "softmax"))
nadam  =  optimizers.Nadam(lr=0.002, beta_1=0.9, beta_2=0.999, epsilon=1e-08, schedule_decay=0.004)
model.compile(loss='categorical_crossentropy', optimizer=nadam, metrics=['accuracy'])

**Comment by enggrey** [comment]
Could you post the code you use to train/evaluate the model?
